### PR TITLE
Add magento.cart connector

### PR DESCRIPTION
**Pull request by SanduDS**
# Description

add Magento cart connector

Fixes https://github.com/ballerina-platform/ballerina-extended-library/issues/226

One line release note: 
- add Magento cart connector

## Type of change

- [X] New feature (non-breaking change which adds functionality)

**Test Configuration**:
* Ballerina Version: Ballerina Swan Lake Beta 6
* Operating System: Linux 
* Java SDK: Java 11

# Checklist:

### Changes to OpenAPI definition

- [X] Modified `info` section of the OpenAPI definition file - [Example](https://github.com/ballerina-platform/ballerina-extended-library/discussions/74)

    - `description` - what the connector is about 
    - `x-ballerina-init-description` OpenAPI extension - connector initialization details 
    - `x-ballerina-display` OpenAPI extension - connector name and path to the connector icon 

- [X] Added proper description each API Key, if API Key authentication is defined in OpenAPI definition file 

### Changes to connector module

- [X] Added license header at the top of each .bal file. 
- [X] Added Package.md as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/77)
- [X] Added Module.md for the module of the  connector as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/78)
- [X] Added `icon.png`
- [X] Added Ballerina.toml file with following information. For keywords refer to [guide](https://github.com/ballerina-platform/ballerina-extended-library/discussions/72)

### Verifying connector module

- [X] Compiled the connector successfully with the targeted ballerina version. 
- [] Conducted smoke tests on the connector (Recommended when OpenAPI definition is obtained from an unofficial source)

### Security checks
 - [X] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [X] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: openapi/magento.cart/Ballerina.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[package]
+license = ["Apache-2.0"]
+keywords = ["Commerce/eCommerce", "Cost/Paid"]
+org = "ballerinax"
+name = "magento.cart"
+distribution = "slbeta6"
+repository = "https://github.com/ballerina-platform/ballerinax-openapi-connectors"
+version = "0.1.0"
```
Comment (RolandHewage) [inline openapi/magento.cart/Ballerina.toml:8]: Shall we update this version to 1.0.0 & build again

File: openapi/magento.cart/client.bal [hunk 2]
```diff
@@ -0,0 +1,820 @@
...
+    # + return - An error if connector initialization failed 
+    public isolated function init(ClientConfig clientConfig, string serviceUrl) returns error? {
+        http:Client httpEp = check new (serviceUrl, clientConfig);
+        self.clientEp = httpEp;
+        return;
+    }
+    # carts/
+    #
+    # + return - 200 Success. 
+    remote isolated function quoteCartManagementV1CreateEmptyCartPost() returns int|error {
```
Comment (RolandHewage) [inline openapi/magento.cart/client.bal:73]: Please if possible update the method names in a meaningful way in all the possible scenarios.

File: openapi/magento.cart/client.bal [hunk 3]
```diff
@@ -0,0 +1,820 @@
...
+    #
+    # + clientConfig - The configurations to be used when initializing the `connector` 
+    # + serviceUrl - URL of the target service 
+    # + return - An error if connector initialization failed 
+    public isolated function init(ClientConfig clientConfig, string serviceUrl) returns error? {
+        http:Client httpEp = check new (serviceUrl, clientConfig);
+        self.clientEp = httpEp;
+        return;
+    }
+    # carts/
```
Comment (RolandHewage) [inline openapi/magento.cart/client.bal:70]: Please provide meaningful doc comments if possible

**Review by RolandHewage — COMMENTED** [review]
Requested few changes

File: openapi/magento.cart/client.bal [hunk 4]
```diff
@@ -0,0 +1,820 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (sachinira) [inline openapi/magento.cart/client.bal:1]: Change the year to 2022. Check all possible places

**Comment by LakshanSS** [comment]
PR check is failing because Dependecies.toml is generated with `SLBeta6` pack.
Please generate with latest pack and update the PR.

File: openapi/magento.cart/Ballerina.toml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+[package]
+license = ["Apache-2.0"]
+keywords = ["Commerce/eCommerce", "Cost/Paid"]
+org = "ballerinax"
+name = "magento.cart"
+distribution = "slbeta6"
+repository = "https://github.com/ballerina-platform/ballerinax-openapi-connectors"
+version = "1.0.0"
+authors = ["Ballerina"]
```
Comment (RolandHewage) [inline openapi/magento.cart/Ballerina.toml:9]: Please add the icon field

**Review by aneeshafedo — APPROVED** [review]

**Pull request merged**